src/mesh/Utils.hpp

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <mesh/Edge.hpp>
+#include <mesh/Mesh.hpp>
+#include <utility>
+
+namespace precice {
+namespace mesh {
+
+/** return a pointer to the shared vertex of 2 edges
+ * @param[in] a pointer to Edge a
+ * @param[in] a pointer to Edge a
+ * @returns a pointer to the Vertex shared by a and b or nullptr otherwise.
+ */
+inline Vertex *sharedVertex(Edge &a, Edge &b)
+{
+  Vertex *a0 = &a.vertex(0);
+  Vertex *a1 = &a.vertex(1);
+  Vertex *b0 = &b.vertex(0);
+  Vertex *b1 = &b.vertex(1);
+  if (a0 == b0 || a0 == b1) {
+    return a0;
+  }
+  if (a1 == b0 || a1 == b1) {
+    return a1;
+  }
+  return nullptr;
+}
```

Comment:
What if both edges are identical (same start and end vertex)?

Reply:
Still returns one shared vertex.
This is valid behaviour, will document it.